### PR TITLE
Adds the initial FAQ documentation for Hydrator UI

### PR DESCRIPTION
**Pull request by ajainarayanan**
Docs-quick-Build: http://builds.cask.co/browse/CDAP-DOB95

### CODE REVIEW

File: cdap-docs/faqs/source/hydrator.rst [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+What should I do when I see the following errors in User Interface?
+-------------------------------------------------------------------
+
+Error parsing widgets JSON for the plugin
+.........................................
+
+This error means the widget JSON for the particular plugin has error in its JSON. The most likely case why this error happens could be because one of the following reasons,
+
+  - You are missing a comma (',') in your JSON
+  - You have a missing double quotes ('"') in you JSON
```
Comment (anwar6953) [inline cdap-docs/faqs/source/hydrator.rst]: your JSON

File: cdap-docs/faqs/source/hydrator.rst [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+----------------------------------------------------------------------------------
+
+The properties are specified as a list inside the configuration groups. So the properties of the plugin will appear inside the group in the same order as they are listed.
+
+What happens if I don't have a property of the plugin in Widget JSON?
+---------------------------------------------------------------------
+
+The Hydrator UI will create a separate group named 'Generic' and adds all properties as part of that particular group and defaults all properties to textbox widget.
+
+What happens when I use an invalid widget (non-existant) in Widget JSON for a particular property in a
```
Comment (anwar6953) [inline cdap-docs/faqs/source/hydrator.rst]: nonexistent

**Comment by awholegunch** [comment]
I've updated the PR with commits. Please check them carefully!

**Comment by awholegunch** [comment]
I've fixed the formatting errors, and this passes a [Quick Build](http://builds.cask.co/browse/CDAP-DOB95-8).

[Hydrator FAQ page](http://builds.cask.co/artifact/CDAP-DOB95/shared/build-8/Docs-HTML/3.3.1-SNAPSHOT/en/faqs/hydrator.html)

File: cdap-docs/faqs/source/hydrator.rst [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+----------------------------------------------------------------------------------
+- **Error parsing widgets JSON for the plugin <plugin-name>**
+
+  This error means the widget JSON for the particular plugin has an error in its JSON, such as: 
+
+  - You are missing a comma character (``,``) in the JSON
+  - You have a missing a double quote character (``"``) in the JSON
+  - You have a value for a key that is not a valid JSON value (for instance, ``NaN``)
+
+  The fastest way to fix these issues is to run the plugin's Widget JSON through a JSON linter
```
Comment (ajainarayanan) [inline cdap-docs/faqs/source/hydrator.rst]: is to use json-lint to identify problems and make sure the widget JSON is valid.

File: cdap-docs/faqs/source/hydrator.rst [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+===================
+FAQs: Cask Hydrator
+===================
+
+What should I do if I see any of these errors in the Cask Hydrator User Interface?
+----------------------------------------------------------------------------------
+- **Error parsing widgets JSON for the plugin <plugin-name>**
+
+  This error means the widget JSON for the particular plugin has an error in its JSON, such as:
```
Comment (ajainarayanan) [inline cdap-docs/faqs/source/hydrator.rst]: Link to widget JSON documentation. Also call the destination title as Widget JSON (currently it says Configuration Json)

**Comment by elfenheart** [comment]
:+1: LGTM

**Pull request merged**